### ISSUE TITLE
Segmentation Fault while importing sparseconvnet in python

### ISSUE THREAD

**Issue by ankita-kalra** [facebookresearch/SparseConvNet#63]
I am unable to run `examples/hello-world.py`. 
It gives me Segmentation Fault (core dumped) in line 9 which is:
`import sparseconvnet as scn`

**Comment by ankita-kalra** [comment]
Backtrace after the Segmentation Fault: 
```
 0x00007fffb3a49a1b in pybind11::detail::make_new_python_type (rec=...) at /home/akalra1/anaconda3/lib/python3.6/site-packages/torch/lib/include/pybind11/detail/class.h:564
#1  0x00007fffb3a6dce9 in pybind11::detail::generic_type::initialize (this=this@entry=0x7fffffff8f90, rec=...)
    at /home/akalra1/anaconda3/lib/python3.6/site-packages/torch/lib/include/pybind11/pybind11.h:887
#2  0x00007fffb3a6e294 in pybind11::class_<Metadata<1>>::class_<>(pybind11::handle, char const*) (this=0x7fffffff8f90, scope=..., name=0x7fffb3b0e709 "Metadata_1")
    at /home/akalra1/anaconda3/lib/python3.6/site-packages/torch/lib/include/pybind11/pybind11.h:1065
#3  0x00007fffb3a6e44d in dimension<1> (m=..., name=name@entry=0x7fffb3b0e709 "Metadata_1") at sparseconvnet/SCN/pybind.cpp:12
#4  0x00007fffb3a226b0 in pybind11_init_SCN (m=...) at sparseconvnet/SCN/pybind.cpp:203
#5  PyInit_SCN () at sparseconvnet/SCN/pybind.cpp:200
#6  0x00005555557698e5 in _PyImport_LoadDynamicModuleWithSpec ()
#7  0x0000555555769ae5 in _imp_create_dynamic ()
#8  0x0000555555665a61 in PyCFunction_Call ()
#9  0x0000555555719fdb in _PyEval_EvalFrameDefault ()
#10 0x00005555556eba94 in _PyEval_EvalCodeWithName ()
#11 0x00005555556ec941 in fast_function ()
#12 0x00005555556f2755 in call_function ()
#13 0x0000555555714cba in _PyEval_EvalFrameDefault ()
#14 0x00005555556ec70b in fast_function ()
#15 0x00005555556f2755 in call_function ()
#16 0x0000555555714cba in _PyEval_EvalFrameDefault ()
#17 0x00005555556ec70b in fast_function ()
#18 0x00005555556f2755 in call_function ()
#19 0x0000555555714cba in _PyEval_EvalFrameDefault ()
#20 0x00005555556ec70b in fast_function ()
#21 0x00005555556f2755 in call_function ()
#22 0x0000555555714cba in _PyEval_EvalFrameDefault ()
#23 0x00005555556ec70b in fast_function ()
#24 0x00005555556f2755 in call_function ()
#25 0x0000555555714cba in _PyEval_EvalFrameDefault ()
#26 0x00005555556ecd7b in _PyFunction_FastCallDict ()
#27 0x0000555555662f5f in _PyObject_FastCallDict ()
#28 0x00005555556a7670 in _PyObject_CallMethodIdObjArgs ()
#29 0x0000555555659a70 in PyImport_ImportModuleLevelObject ()
#30 0x0000555555717033 in _PyEval_EvalFrameDefault ()
#31 0x00005555556ed459 in PyEval_EvalCodeEx ()
#32 0x00005555556ee1ec in PyEval_EvalCode ()
#33 0x00005555557125bd in builtin_exec ()
#34 0x0000555555665a61 in PyCFunction_Call ()
#35 0x0000555555719fdb in _PyEval_EvalFrameDefault ()
#36 0x00005555556eba94 in _PyEval_EvalCodeWithName ()
#37 0x00005555556ec941 in fast_function ()
#38 0x00005555556f2755 in call_function ()
#39 0x0000555555714cba in _PyEval_EvalFrameDefault ()
#40 0x00005555556ec70b in fast_function ()
#41 0x00005555556f2755 in call_function ()
#42 0x0000555555714cba in _PyEval_EvalFrameDefault ()
#43 0x00005555556ec70b in fast_function ()
#44 0x00005555556f2755 in call_function ()
#45 0x0000555555714cba in _PyEval_EvalFrameDefault ()
#46 0x00005555556ec70b in fast_function ()
#47 0x00005555556f2755 in call_function ()
#48 0x0000555555714cba in _PyEval_EvalFrameDefault ()
#49 0x00005555556ecd7b in _PyFunction_FastCallDict ()
#50 0x0000555555662f5f in _PyObject_FastCallDict ()
#51 0x00005555556a7670 in _PyObject_CallMethodIdObjArgs ()
#52 0x0000555555659a70 in PyImport_ImportModuleLevelObject ()
#53 0x0000555555717033 in _PyEval_EvalFrameDefault ()
#54 0x00005555556ed459 in PyEval_EvalCodeEx ()
#55 0x00005555556ee1ec in PyEval_EvalCode ()
#56 0x00005555557125bd in builtin_exec ()
#57 0x0000555555665a61 in PyCFunction_Call ()
#58 0x0000555555719fdb in _PyEval_EvalFrameDefault ()
```

**Comment by btgraham** [comment]
Sorry for the delay. What kind of system are you running this on? What compiler is being used?

**Comment by WayneZww** [comment]
Do you currently using gcc version <=4.9?
I used gcc-4.8.5 before and met the same problem, but the bug was fixed after I changed it to gcc-5.2.0

**Comment by dongxiao92** [comment]
I have encountered the same problem. Changing to gcc with the same or higher version than that used to compile pytorch solves it.
